lib/LedgerSMB/Scripts/recon.pm

```diff
@@ -274,10 +275,13 @@ sub _display_report {
 
     $recon->{zero_string} = LedgerSMB::PGNumber->from_input(0)->to_output(money => 1);
 
-    $recon->{statement_gl_calc} = $neg_factor *
-                                    ($recon->{their_total}
+    $recon->{statement_gl_calc} = $is_positive
+                                  ?  ($recon->{their_total}
                                     + $recon->{outstanding_total}
-                                    + $recon->{mismatch_our_total});
+                                    + $recon->{mismatch_our_total})
+                                  : - ($recon->{their_total}
+                                    + $recon->{outstanding_total}
+                                    + $recon->{mismatch_our_total}) ;
```

Comment:
I like this code as an example of how we might want `-1` *not* to be marked as magical: the consequence is code duplication if we don't. I'm afraid the result of this code duplication is that we end up with mismatching implementations.